contracts/votingStrategy/QuadraticFundingStrategy/QuadraticFundingVotingStrategyImplementation.sol

```diff
@@ -87,19 +82,11 @@ contract QuadraticFundingVotingStrategyImplementation is IVotingStrategy, Initia
           _amount
         );
       }
-
-      /// @dev emit event for transfer
-      emit Voted(
-        _token,
-        _amount,
-        voterAddress,
-        _grantAddress,
-        _projectId,
-        _applicationIndex,
-        msg.sender
-      );
     }
 
-    require(msgValue == msg.value, "msg.value does not match vote amount");
+    /// @dev emit event for transfer
+    emit VotesCast(encodedVotes, voterAddress);
```

Comment:
If the encodedVotes data is large or exceeds the maximum event data size limit, it may not be feasible to emit it directly as an event. Ethereum has a maximum event data size limit, and exceeding it can result in transaction failures. 
Maybe for that reason it might make sense to break this into a batch of events.

Reply:
Maybe something like

```
function vote(bytes[] calldata encodedVotes, address voterAddress) external override payable nonReentrant isRoundContract {
    uint256 totalVotes = encodedVotes.length;
    uint256 batchSize = 10; // We can adjust this but what do we set it to ? 

    for (uint256 i = 0; i < totalVotes; i += batchSize) {
        uint256 end = i + batchSize;
        if (end > totalVotes) {
            end = totalVotes;
        }

        bytes[] memory votesBatch = new bytes[](end - i);
        for (uint256 j = i; j < end; j++) {
            votesBatch[j - i] = encodedVotes[j];
        }

        emit VotesCast(votesBatch, voterAddress, msg.sender);
    }
}
```